go/vt/vtgate/evalengine/evalengine.go

```diff
@@ -235,39 +235,27 @@ func compareNumeric(v1, v2 *EvalResult) (int, error) {
 func parseDate(expr *EvalResult) (t time.Time, err error) {
```

Comment:
Unified what this function depends on also into the single parsing logic in the `sqlparser` package.